app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreenViewModel.kt

```diff
@@ -84,39 +108,76 @@ class OtherUserProfileScreenViewModel @Inject constructor(
     init {
         state = state.copy(isDataLoading = true)
         viewModelScope.launch {
-            when (val result = getUserInfo(userId)) {
-                is GetUserInfoResult.Failure -> {
-                    appLogger.d("Couldn't not find the user with provided id:$userId.id and domain:$userId.domain")
+            when (val conversationResult = getOrCreateOneToOneConversation(userId)) {
+                is CreateConversationResult.Failure -> {
+                    appLogger.d("Couldn't not getOrCreateOneToOneConversation for user id: $userId")
                     showInfoMessage(InfoMessageType.LoadUserInformationError)
                 }
-                is GetUserInfoResult.Success ->
-                    conversationId
+                is CreateConversationResult.Success -> {
+                    val observeConversationRoleData = conversationId
                         .let { if (it != null) observeConversationRoleForUser(it, userId) else flowOf(it) }
-                        .collect { loadViewState(result.otherUser, result.team, it) }
+
+                    combine(observeConversationRoleData, observeSelfUser(), observeUserInfo(), ::Triple)
+                        .collect { (conversationRoleData, selfUser, userInfoResult) ->
+                            loadViewState(userInfoResult, conversationResult.conversation, conversationRoleData, selfUser.teamId)
+                        }
+                }
             }
         }
     }
 
-    private fun loadViewState(otherUser: OtherUser, team: Team?, conversationRoleData: ConversationRoleData?) {
-        state = state.copy(
-            isDataLoading = false,
-            userAvatarAsset = otherUser.completePicture?.let { pic -> ImageAsset.UserAvatarAsset(wireSessionImageLoader, pic) },
-            fullName = otherUser.name ?: String.EMPTY,
-            userName = mapUserLabel(otherUser),
-            teamName = team?.name ?: String.EMPTY,
-            email = otherUser.email ?: String.EMPTY,
-            phone = otherUser.phone ?: String.EMPTY,
-            connectionStatus = otherUser.connectionStatus,
-            membership = userTypeMapper.toMembership(otherUser.userType),
-            groupState = conversationRoleData?.userRole?.let { userRole ->
-                OtherUserProfileGroupState(
-                    groupName = conversationRoleData.conversationName,
-                    role = userRole,
-                    isSelfAnAdmin = conversationRoleData.selfRole is Member.Role.Admin
+    private fun observeUserInfo(): Flow<GetUserInfoResult.Success> =
```

Comment:
I believe this should be in kalium, we should have the `ObserveUserInfoUseCase` which returns `Flow`, it may be useful and this way there would be no need to make a workaround by using `_refreshUserInfo` 🙂

Reply:
yeah, good point :)